fuzz/ml-kem.c

```diff
@@ -0,0 +1,685 @@
+/*
+ * Copyright 2025 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * https://www.openssl.org/source/license.html
+ * or in the file LICENSE in the source distribution.
+ */
+
+/*
+ * Test hashtable operation.
+ */
+#include <string.h>
+#include <openssl/evp.h>
+#include <openssl/err.h>
+#include <openssl/rand.h>
+#include "internal/nelem.h"
+#include "fuzzer.h"
+
+/**
+ * @brief Consumes a 16-bit unsigned integer from a buffer.
+ *
+ * This function extracts a 16-bit unsigned integer from the given buffer
+ * while ensuring proper alignment. It adjusts the buffer pointer to the
+ * next 16-bit boundary, extracts the value, advances the pointer, and
+ * updates the remaining buffer length.
+ *
+ * @param buf  Pointer to the input buffer.
+ * @param len  Pointer to the size of the remaining buffer; updated after consumption.
+ * @param val  Pointer to store the extracted 16-bit value.
+ *
+ * @return Pointer to the updated buffer position after reading the value,
+ *         or NULL if the buffer does not contain enough data.
+ */
+static uint8_t *consume_uint16t(const uint8_t *buf, size_t *len, uint16_t *val)
```

Comment:
Is of this redundant given `<openssl/byte_order.h>`?

Reply:
I honestly hadn't thought about using OPENSSL_load_* apis to do this, but it does look like it would work, and save some code.  Thats a good suggestion, I'll make the change.